osquery/database/database.cpp

```diff
@@ -366,16 +366,31 @@ Status ptreeToRapidJSON(const std::string& in, std::string& out) {
     return Status(1, "Failed to parse JSON");
   }
 
+  if (tree.empty()) {
+    JSON().toString(out);
+    return Status();
+  }
+
   auto json = JSON::newArray();
```

Comment:
I don't think this works for instance for `{ "v1" : "k1" , "v2" : "k2" }`. This shouldn't be an array but rather an object.

Reply:
Sure, I think the reason we had done this was largely because this upgrade function wasn't intended to perfectly convert a ptree to an RJ, it was just meant to handle the edge cases we were seeing in the upgrade path where lists in ptrees were not correct list json types.